.github/workflows/ci.yml

```diff
@@ -0,0 +1,435 @@
+name: Alfresco Community Repo CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - feature/**
+      - fix/**
+  workflow_call:
+
+env:
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  GITHUB_ACTIONS_DEPLOY_TIMEOUT: 60
+  JAVA_VERSION: "17"
+  LOG_WARN: "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  QUAY_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
+  QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
+  TAS_ENVIRONMENT: ./packaging/tests/environment
+  TAS_SCRIPTS: ../alfresco-community-repo/packaging/tests/scripts
+
+jobs:
+  prepare:
+    name: "Prepare"
+    runs-on: ubuntu-latest
+    if: >
+      ! contains(github.event.head_commit.message, '[skip tests]') &&
+      ! contains(github.event.head_commit.message, '[force]')
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Clean cache"
+        run: bash ./scripts/travis/cleanup_cache.sh
+      - name: "Prepare environment"
+        run: bash ./scripts/travis/prepare.sh
```

Comment:
these scripts will always work on an empty .m2 folder if cache is not restored before

If I understood correctly what they are for, you probably need to author a custom composite action to reuse this steps (along with cache restore) in any job that would invoke maven

this job as it is can be removed